state/application.go

```diff
@@ -378,14 +338,13 @@ func (op *DestroyApplicationOperation) destroyOps(store objectstore.ObjectStore)
 	// If the application has no units, and all its known relations will be
 	// removed, the application can also be removed, so long as there are
 	// no other cluster resources, as can be the case for k8s charms.
-	if op.app.doc.UnitCount == 0 && op.app.doc.RelationCount == removeCount {
+	if op.app.doc.UnitCount == 0 {
```

Comment:
todo: delete `RelationCount` from the applicationDoc struct. It's still used in the DestroyApplicationOperation.